src/utils/SpokeUtils.ts

```diff
@@ -42,13 +41,15 @@ export async function getBlockRangeForDepositId(
   // in the queriedIds cache.
   const _getDepositIdAtBlock = async (blockNumber: number): Promise<number> => {
     if (queriedIds[blockNumber] === undefined) {
-      queriedIds[blockNumber] = await getDepositIdAtBlock(spokePool, blockNumber);
+      const resultId = await getDepositIdAtBlock(spokePool, blockNumber);
+      if (!Number.isInteger(resultId)) {
+        throw new Error("Invalid deposit count");
+      }
```

Comment:
Would it be better to locate this validation centrally in `getDepositIdAtBlock()`, so callers don't have to check this every time?